File: app/src/main/java/com/example/forestgame/SettingsActivity.java
package com.example.forestgame;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class SettingsActivity extends AppCompatActivity {
    public static final String PREFS_NAME = "forest_prefs";
    public static final String KEY_QUALITY = "quality";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        Spinner spinner = findViewById(R.id.qualitySpinner);
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
                this,
                R.array.quality_options,
                android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);

        SharedPreferences preferences = getSharedPreferences(PREFS_NAME, MODE_PRIVATE);
        int currentQuality = preferences.getInt(KEY_QUALITY, 1);
        spinner.setSelection(currentQuality);

        spinner.setOnItemSelectedListener(new SimpleOnItemSelectedListener(position ->
                preferences.edit().putInt(KEY_QUALITY, position).apply()));
    }
}

File: app/src/main/java/com/example/forestquest/SettingsActivity.kt
package com.example.forestquest

import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import com.example.forestquest.databinding.ActivitySettingsBinding

class SettingsActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.toolbar.setNavigationOnClickListener { finish() }

        val labels = GraphicsQuality.labels()
        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, labels)
        binding.qualitySpinner.setAdapter(adapter)

        val currentQuality = GraphicsPreferences.getQuality(this)
        binding.qualitySpinner.setText(currentQuality.toLabel(), false)

        binding.qualitySpinner.setOnItemClickListener { _, _, position, _ ->
            val selected = GraphicsQuality.fromLabel(labels[position])
            GraphicsPreferences.saveQuality(this, selected)
        }
    }

    private fun GraphicsQuality.toLabel(): String = when (this) {
        GraphicsQuality.HIGH -> "Высокое"
        GraphicsQuality.MEDIUM -> "Среднее"
        GraphicsQuality.LOW -> "Низкое"
    }
}
